internal/cmd/local/local/install.go

```diff
@@ -380,43 +406,23 @@ func (c *Command) watchEvents(ctx context.Context) {
 	}
 }
 
-// 2024-09-10 20:16:24 WARN i.m.s.r.u.Loggers$Slf4JLogger(warn):299 - [273....
-var javaLogRx = regexp.MustCompile(`^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} \x1b\[(?:1;)?\d+m(?P<level>[A-Z]+)\x1b\[m (?P<msg>\S+ - .*)`)
-
 func (c *Command) streamPodLogs(ctx context.Context, namespace, podName, prefix string, since time.Time) error {
 	r, err := c.k8s.StreamPodLogs(ctx, namespace, podName, since)
 	if err != nil {
 		return err
 	}
 	defer r.Close()
 
-	level := pterm.Debug
-	scanner := bufio.NewScanner(r)
-
-	for scanner.Scan() {
-
-		// skip java stacktrace noise
-		if strings.HasPrefix(scanner.Text(), "\tat ") || strings.HasPrefix(scanner.Text(), "\t... ") {
-			continue
-		}
-
-		m := javaLogRx.FindSubmatch(scanner.Bytes())
-		var msg string
-
-		if m != nil {
-			msg = string(m[2])
-			if string(m[1]) == "ERROR" {
-				level = pterm.Error
-			} else {
-				level = pterm.Debug
-			}
+	s := newJavaLogScanner(r)
```

Comment:
It is true that *most* of the pods are running Java services, but not all. Just wanna make note of that in case we ever try to use this to pull errors from temporal or something.

Reply:
Ya, the log parsing stuff is fragile right now. And maybe I should remove the `java` part of the name. It's really just a log scanner for a certain format.

Reply:
Added tests and cleaned up the naming to not mention java